public/src/client.ts

```diff
@@ -0,0 +1,121 @@
+import { CallOptions, ConnectError, Transport, makeAnyClient } from "@bufbuild/connect";
+import { createAsyncIterable } from "@bufbuild/connect/protocol";
+import {
+    MethodInfo,
+    MethodInfoServerStreaming,
+    MethodInfoUnary,
+    PartialMessage,
+    ServiceType,
+    Message,
+    MethodKind,
+    AnyMessage,
+} from "@bufbuild/protobuf";
+
+export type Response<T extends AnyMessage> = {
+    error: ConnectError | null
+    message: T
+}
+
+/**
+ * ResponseClient is a simple client that supports unary and server-streaming
+ * methods. Methods will produce a promise for the response message,
+ * or an asynchronous iterable of response messages.
+ */
+export type ResponseClient<T extends ServiceType> = {
```

Comment:
Not sure it makes sense, but is there a test case for this stuff that we could add to show how this is actually used.